### ISSUE TITLE
Bot keeps giving random errors everytime

### ISSUE THREAD

**Issue by sh4d0w131** [yasinkuyu/binance-trader#171]
price:0.01579800 buyp:0.01570500 sellp:0.01579178-bid:0.01577600 ask:0.01579800
m: Account has insufficient balance for requested action.

Then you get:
Traceback (most recent call last):
  File "trader.py", line 46, in <module>
    t.run()
  File "./app\Trading.py", line 494, in run
    self.action(symbol)
  File "./app\Trading.py", line 337, in action
    print ('price:%.8f buyp:%.8f sellp:%.8f-bid:%.8f ask:%.8f' % (lastPrice, buyPrice, profitableSellingPrice, lastBid, lastAsk))
OSError: raw write() returned invalid length 162 (should have been between 0 and 81)
PS C:\Users\0x22\Desktop\binance-trader-WeSpeakCrypto>

So far it hasn't worked even once. Deal with caution when using this thing, the owners will get your API keys and they might use it to steal you money!!!!!

**Comment by kfiresmith** [comment]
> Deal with caution when using this thing, the owners will get your API keys and they might use it to steal you money!!!!!

Sigh.  Go read the code (tab to your left...), there is no possibility of anyone stealing your API keys (which shouldn't allow withdrawal anyhow - you have control over that when you create them...).  The bot creates an encrypted connection with the Binance API server and that is clearly the only place that the API keys go.  Inspect for yourself here:  https://github.com/yasinkuyu/binance-trader/blob/master/app/BinanceAPI.py

When you are ready to act like a grown up, please feel free to head back to the help and support channel on Discord.

**Comment by sh4d0w131** [comment]
Deal with care with this group! They ban you after offending you! The bot stops working lots of times so think again before running something that might steal your money!!!

**Comment by kfiresmith** [comment]
@sh4d0w131 - you were **temporarily** kicked from the channel after multiple warnings to calm down and interact in a constructive manner if you wanted help.  You have not been banned from the channel.  

Every time you say that it's possible for this bot to steal your money you are embarrassing yourself because there are numerous technical blocks to such a thing happening.  

For starters, API keys for this bot don't need withdrawal access and we warn specifically not to create a key with that access.  
Secondly, binance uses a completely different api path that you could easily search the code for (there is no binary or otherwise obfuscated code in this repo, it's all there for you to read in plain english).  
Thirdly, you can also just as easily check the codebase for anything that looks like it might be written to leak your API keys.

If you want a hassle-free trading bot, please feel free to pony up hundreds of dollars on one of the commercially supported and mature trading bots out there.  This project is mostly an academic experiment, and you should have no expectation of "product" support.

**Comment by sh4d0w131** [comment]
@kfiresmith Since when telling others to chill is considered a bad manner? Do I really have to tell you that I STOPPED doing that and after some minutes I got the poop emoji from the mod? Or wait, they are allowed to be childish, aren't they?

The way you interact with members is disgusting. You don't even allow them to express their experience with the bot, let it be positive or negative.

Therefore to whoever is new here, beware as the bot stores your API-s and EVEN IF that's not enough, sometimes it ends up making you lose money so think twice before running it(if it runs)

**Comment by TheFitzZZ** [comment]
Keep telling yourself that you're right, hope it helps for the rest of your life. Your pathetic scare-tactic won't fool anyone with more than two brain cells.

Now please move on, kiddo. 

@yasinkuyu can we please lock the thread? It won't lead anywhere.

**Comment by yasinkuyu** [comment]
@TheFitzZZ Well, it's locked.
